mm-go-irckit/server_commands.go

```diff
@@ -449,29 +456,51 @@ func parseModifyMsg(u *User, msg *irc.Message, channelID string) bool {
 		text = matches[2]
 	}
 
-	id, err := strconv.ParseInt(matches[1], 16, 0)
-	if err != nil {
-		logger.Errorf("couldn't parseint %s: %s", matches[1], err)
-	}
+	msgID := ""
 
-	u.msgMapMutex.RLock()
-	defer u.msgMapMutex.RUnlock()
+	switch {
+	// last message from user. Also support '!!' like shell's history
+	// substitution for previous command.
+	case matches[1] == "//" || matches[1] == "/!!/":
+		u.msgLastMutex.RLock()
+		defer u.msgLastMutex.RUnlock()
+		msgLast, ok := u.msgLast[channelID]
+		if ok {
+			msgID = msgLast
+		}
+	// Mattermost message/thread ID (e.g. 'cfrakpwix7y8pgzux6ta76pm9c')
+	case len(matches[1]) == 28:
+		msgID = strings.ReplaceAll(matches[1], "/", "")
+		u.msgLastMutex.Lock()
+		defer u.msgLastMutex.Unlock()
+		u.msgLast[channelID] = msgID
+	// matterircd message/thread ID (e.g. '004' and 'a12')
+	case len(matches[1]) == 5:
+		id, err := strconv.ParseInt(strings.ReplaceAll(matches[1], "/", ""), 16, 0)
+		if err != nil {
+			logger.Errorf("couldn't parseint %s: %s", matches[1], err)
+		}
 
-	m := u.msgMap[channelID]
+		u.msgMapMutex.RLock()
+		defer u.msgMapMutex.RUnlock()
 
-	msgID := ""
+		m := u.msgMap[channelID]
 
-	for k, v := range m {
-		if v == int(id) {
-			msgID = k
+		for k, v := range m {
+			if v == int(id) {
+				msgID = k
+				u.msgLastMutex.Lock()
+				defer u.msgLastMutex.Unlock()
+				u.msgLast[channelID] = msgID
+			}
```

Comment:
```suggestion
			if v != int(id) {
				continue
			}
			
			msgID = k
			
			u.msgLastMutex.Lock()
			defer u.msgLastMutex.Unlock()
			
			u.msgLast[channelID] = msgID
```

Original loop was only one statement, now we have more statement this is better.